Index: git/BrowserLauncher/src/wpewebkit/wpewebkitconfig.cpp
===================================================================
--- git.orig/BrowserLauncher/src/wpewebkit/wpewebkitconfig.cpp
+++ git/BrowserLauncher/src/wpewebkit/wpewebkitconfig.cpp
@@ -61,6 +61,12 @@ unsigned long readLimits(const std::stri
 void setEnvVar(const char *varName, const std::string& value, bool replace)
 {
     // FIXME: setenv/putenv are not thread safe and may cause random crashes.
+    if (const char *env = g_getenv(varName); (env != nullptr) && (replace == false)) {
+        g_message(">%s = \"%s\"", varName, env);
+    } else {
+        g_message(" %s = \"%s\"", varName, value.c_str());
+    }
+
     g_setenv(varName, value.c_str(), replace);
 }
 
